### PR TITLE
Winner

### LINKED ISSUE #26: IN: Selecting a winner
\- When the contest is over
  \- no more submissions
  \- As the contest owner, at the end of the contest deadline, you can choose a winner 
    \- Confirm the winner 
\- Mark that submission as winner
\- Payment goes through 
\- One restriction: CC must be entered before a contest is created

### PR DESCRIPTION
**Pull request by HAZberi**
### What this PR does (required):
- Selecting the Winner
- Charge Customer
- Save Customer to Retrieve on Discover Page
- UI Enhancement
- Corrected User flow aka user journey

### Screenshots / Videos (required):
- N/A

### Any information needed to test this feature (required):
- Run the project in your local browser to test user journey.

### Any issues with the current functionality (optional):
- Email system is not integrated.

### CODE REVIEW

File: client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx [hunk 1]
```diff
@@ -9,50 +9,72 @@ import Button from '@material-ui/core/Button';
...
+  const { updateSnackBarMessage } = useSnackBar();
 
   if (loggedInUser === undefined || !loggedInUser || contest == undefined || contest.userID == undefined)
     return <CircularProgress />;
 
+  const handleWinnerSelection = async (
+    imageSrc: string,
+    author: User,
+    contest: Contest,
+    user: User | null | undefined,
```
Comment (rajivtitus) [inline client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx:34]: Why are we using null/undefined types here?

File: client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx [hunk 2]
```diff
@@ -9,50 +9,72 @@ import Button from '@material-ui/core/Button';
...
   return (
     <Card className={classes.submissionCard}>
       <CardMedia component="img" alt="Contemplative Reptile" height="300" image={imageSrc} />
       <Box className={classes.displayOver}>
         <CardContent className={classes.submissionContent}>
           <Typography gutterBottom variant="h5" component="h2">
-            {loggedInUser.id == contest?.userID._id && (
-              <Button className={classes.winner}>
+            {deadlinePassed(contest.deadlineDate) ? (
+              <Button
```
Comment (rajivtitus) [inline client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx:65]: Is the Winner button currently visible to all users?

**Review by rajivtitus — APPROVED** [review]
LGTM! Just a couple of minor comments

File: client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx (hunk 1 above)
Comment (HAZberi) [reply client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx:34]: user types are coming from the loggedinUser. If I just change it to user. It's throwing type error.

File: client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx (hunk 2 above)
Comment (HAZberi) [reply client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx:65]: Winner button will only be visible to the contest creator on his dashboard under completed contest, once the deadline is passed.

**Comment by HAZberi** [comment]
request changes implemented successfully.

**Pull request merged**